include/cpp_bindgen/common/disjunction.hpp

```diff
@@ -0,0 +1,22 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include <type_traits>
+
+namespace cpp_bindgen {
+
+    template <class...>
+    struct disjunction : std::false_type {};
+    template <class B1>
+    struct disjunction<B1> : B1 {};
+    template <class B1, class... Bn>
+    struct disjunction<B1, Bn...> : std::conditional<bool(B1::value), B1, disjunction<Bn...>>::type {};
```

Comment:
This is how it's defined in the standard. B1 is not required to be `std::true_type`, and it will just return the first type, for which `bool(B::value)` is true. It doesn't matter in this case, but disjunction is part of the standard starting with C++17, so why not take the definition from there.

Reply:
sure, didn't know it is from there.